Lib/test/test_ordered_dict.py

```diff
@@ -729,6 +730,22 @@ def test_merge_operator(self):
         with self.assertRaises(ValueError):
             a |= "BAD"
 
+    def test_unstable_hash_should_not_abort_issue132461(self):
+        # OrderedDict should raise, not abort, when used with unstable-hash keys
+        large_num = 2**64
+
+        class WeirdBase(abc.ABCMeta):
+            def __hash__(self):
+                return randint(0, large_num)
+
+        class weird_bytes(bytes, metaclass=WeirdBase):
+            pass
+
+        obj = self.OrderedDict()
+        with self.assertRaises(Exception):
+            for x in range(100):
+                obj.setdefault(weird_bytes, None)
```

Comment:
```suggestion
        class A:
            c = 0
            def __hash__(self):
                self.c += 1
                return self.c

        a = A()
        obj = self.OrderedDict()
        msg = "OrderedDict key appears to change its hash value over time"
        with self.assertRaisesRegex(RuntimeError, msg):
            for _ in range(100):
                obj.setdefault(a, None)
```

We can also remove `abc` dependency and `randint` import.

Reply:
part of test
        with self.assertRaises(KeyError):
            for x in range(100):
                obj.setdefault(weird_bytes, None)

error
```
ERROR: test_unstable_hash_should_not_abort_issue132461 (test.test_ordered_dict.CPythonOrderedDictTests.test_unstable_hash_should_not_abort_issue132461)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/learning/cpython/Lib/test/test_ordered_dict.py", line 747, in test_unstable_hash_should_not_abort_issue132461
    obj.setdefault(weird_bytes, None)
    ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^
RuntimeError: OrderedDict key appears to change its hash value over time
```

fixed part of test
```
        with self.assertRaises(RuntimeError):
            for x in range(100):
                obj.setdefault(weird_bytes, None)
```


ERROR: test_unstable_hash_should_not_abort_issue132461 (test.test_ordered_dict.CPythonOrderedDictSubclassTests.test_unstable_hash_should_not_abort_issue132461)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/learning/cpython/Lib/test/test_ordered_dict.py", line 747, in test_unstable_hash_should_not_abort_issue132461
    obj.setdefault(weird_bytes, None)
    ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^
KeyError: <class 'test.test_ordered_dict.OrderedDictTests.test_unstable_hash_should_not_abort_issue132461.<locals>.weird_bytes'>

======================================================================
ERROR: test_unstable_hash_should_not_abort_issue132461 (test.test_ordered_dict.PurePythonOrderedDictSubclassTests.test_unstable_hash_should_not_abort_issue132461)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/learning/cpython/Lib/test/test_ordered_dict.py", line 747, in test_unstable_hash_should_not_abort_issue132461
    obj.setdefault(weird_bytes, None)
    ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^
  File "/learning/cpython/Lib/collections/__init__.py", line 274, in setdefault
    return self[key]
           ~~~~^^^^^
KeyError: <class 'test.test_ordered_dict.OrderedDictTests.test_unstable_hash_should_not_abort_issue132461.<locals>.weird_bytes'>

======================================================================
ERROR: test_unstable_hash_should_not_abort_issue132461 (test.test_ordered_dict.PurePythonOrderedDictTests.test_unstable_hash_should_not_abort_issue132461)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/learning/cpython/Lib/test/test_ordered_dict.py", line 747, in test_unstable_hash_should_not_abort_issue132461
    obj.setdefault(weird_bytes, None)
    ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^
  File "/learning/cpython/Lib/collections/__init__.py", line 274, in setdefault
    return self[key]
           ~~~~^^^^^
KeyError: <class 'test.test_ordered_dict.OrderedDictTests.test_unstable_hash_should_not_abort_issue132461.<locals>.weird_bytes'>
```


Very strange

Reply:
Still, we should only catch `RuntimeError` and not arbitrary exceptions. The assertion failing may be an indication that something else is wrong.